### ISSUE TITLE
Allow users to configure the per-health-check-timeout

### ISSUE THREAD

**Issue by LeePorte** [cloudfoundry/cloud_controller_ng#970]
## Issue

Currently users are unable to set the response timeout on recurring healthchecks.

## Context

We have a user that has a healthcheck that takes ~1300ms to complete on occasion due to other processes accessing the database. Due to this taking over 1000ms the user app fails the healthcheck and as a result is restarted.  

## Steps to Reproduce

Have an app with a healthcheck that takes more than 1000ms to return

## Expected result

With a user configured healthcheck timeout the app would not end up in a restart loop due to healthchecks taking longer than 1000ms to complete. 

## Current result

The user app is restarted

## Possible Fix
A possible fix has been identified by @ljfranklin and @emalm 

```
emalm [5:03 PM] 
@lfranklin there are two timeouts at play: the instance start timeout and the per-health-check timeout

the `health_check_timeout` controls the instance-start timeout (how long an instance has to become healthy before diego considers it to have failed), and the thing we reverted was applying that also to the per-health-check timeout

in general, they are different quantities: you don't necessarily want each health check attempt to be given up to, say, 3 or 10 minutes to time out if that's how long your instances are granted to become healthy

lfranklin [5:08 PM] 
@emalm I see, so we could allow users to configure the `per-health-check-timeout` by setting these fields: https://github.com/cloudfoundry/cloud_controller_ng/blob/master/lib/diego/bbs/models/check_definition.pb.rb#L42-L47. But this timeout would be a new property on the API different from the existing `health_check_timeout` which is the total amount of time an app has to become healthy.

emalm [5:08 PM] 
yep
```

**Comment by zrob** [comment]
Implemented in this [story](https://www.pivotaltracker.com/story/show/156819169).  Available in capi-release [1.54.0](https://github.com/cloudfoundry/capi-release/releases/tag/1.54.0)

Docs [here](http://v3-apidocs.cloudfoundry.org/version/3.43.0/index.html#the-health_check-object)

**Comment by bobbyhubbard** [comment]
Did this get closed on accident? The message from @zrob indicates it was implemented but the story references `invocation_timeout` and not `per-health-check-timeout` as indicated in the possible fix from @ljfranklin. Was it decided to not implement `per-health-check-timeout` as suggested above?
